packages/server/lib/util/socket_allowed.ts

```diff
@@ -19,7 +19,7 @@ export class SocketAllowed {
     const { localPort } = socket
 
     debug('allowing socket %o', { localPort })
-    this.allowedLocalPorts.push(localPort)
+    this.allowedLocalPorts.push(localPort!)
```

Comment:
I see a few places where we do `!`. Should these be handled by throwing an error instead? Eg

```ts
if (!localPort.socket) {
  throw Error(`localPort.socket was not defined. This should never happen`)
}
```

Much easier to track down if we ever hit an otherwise impossible code path.

Reply:
These codes are not run frequently. I guess using condition for Error message can be a good idea. I changed them.